# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/modules/TabUnloader.jsm
# Commit: 812ed5275f0d
# Full Hash: 812ed5275f0db9f92944955f891df2ab68c9f8e7
# Author: Toshihito Kikuchi <tkikuchi@mozilla.com>
# Date: 2021-11-19 04:11:43
# Regressor Bug: 1734099
# File Overlap Count: 5
# Description:
#   Bug 1734099 - Never unload tabs with active RTCPeerConnection instances.  r=peterv,jesup
#   
#   This patch makes sure the Tab Unloading feature does not unload tabs that have
#   active peer connections not to disrupt browsing experience based on WebRTC peer
#   connections.
# ==============================================================================

diff -r a733d049dbda -r 812ed5275f0d browser/modules/TabUnloader.jsm
--- a/browser/modules/TabUnloader.jsm	Fri Nov 19 01:03:56 2021 +0200
+++ b/browser/modules/TabUnloader.jsm	Thu Nov 18 22:59:12 2021 +0000
@@ -81,7 +81,17 @@
   },
 
   usingWebRTC(tab, weight) {
-    return webrtcUI.browserHasStreams(tab.linkedBrowser) ? weight : 0;
+    const browser = tab.linkedBrowser;
+    if (!browser) {
+      return 0;
+    }
+
+    // No need to iterate browser contexts for hasActivePeerConnection
+    // because hasActivePeerConnection is set only in the top window.
+    return webrtcUI.browserHasStreams(browser) ||
+      browser.browsingContext?.topWindowContext?.hasActivePeerConnections
+      ? weight
+      : 0;
   },
 
   getMinTabCount() {